### PR TITLE
Use rs_merkle

### LINKED ISSUE #144: [sBTC DR] Use bitcoin merkle tree instead of stacks merkle tree
The current implementation uses stacks merkle trees with tagged hashes using 0 (leaf) and 1 (path).
For bitcoin tx verification, we need merkle trees without tags.

### PR DESCRIPTION
**Pull request by friedger**
## Description
This PR
* adds rs_merkle
* implements Hasher for double sha256
* fixes tests that does not use tagged data in merkle trees anymore
* fixes #144 

Now we can mint in devenv

Screenshot from 2023-09-18 13-36-50

### CODE REVIEW

**Review by stjepangolemac — APPROVED** [review]
Looks good. I've flagged a couple of things but they're not critical at all.

File: romeo/src/proof_data.rs [hunk 1]
```diff
@@ -1,24 +1,44 @@
...
+    }
+}
 
 /// Data needed to prove that a bitcoin transaction was mined on the bitcoin
 /// network. This data is used by clarity contracts.
 #[derive(Debug, Clone)]
 pub struct ProofData {
-    /// The transaction id of the bitcoin transaction
-    pub txid: BitcoinTxId,
+    /// The reversed transaction id of the bitcoin transaction
```
Comment (jo-tm) [inline romeo/src/proof_data.rs:27]: Can you document why we need the reversed transaction? Is this an issues with the Clarity Language?

File: romeo/src/proof_data.rs [hunk 2]
```diff
@@ -41,26 +61,45 @@ impl ProofData {
     /// Create a new proof from a bitcoin transaction and a block
     pub fn from_block_and_index(block: &Block, index: usize) -> Self {
         let tx: &Transaction = block.txdata.get(index).expect("Invalid tx index");
+        let mut merkle_tree = MerkleTree::<DoubleSha256Algorithm>::new();
+        for tx in &block.txdata {
+            merkle_tree.insert(tx.txid().to_vec().try_into().unwrap());
+        }
+        // append last tx id if number of leaves is odd
+        if block.txdata.len() % 2 == 1 {
```
[the remark this answers is not in the document]
Comment (friedger) [reply romeo/src/proof_data.rs:69]: No

File: romeo/src/proof_data.rs [hunk 3]
```diff
@@ -83,7 +125,7 @@ impl ProofData {
                 data: self
                     .merkle_path
                     .iter()
-                    .map(|v| Value::buff_from(v.hash.to_bytes().to_vec()).unwrap())
+                    .map(|v| Value::buff_from(v.to_vec()).unwrap())
```
[the remark this answers is not in the document]
Comment (friedger) [reply romeo/src/proof_data.rs:128]: :+1:

**Comment by friedger** [comment]
Why is the pr-validation failing? Anything we can do about it?
```
[2023-09-18T15:14:08.986Z] ['error'] There was an error running the uploader: Error uploading to [https://codecov.io:](https://codecov.io/) Error: There was an error fetching the storage URL during POST: 404 - {'detail': ErrorDetail(string='Unable to locate build via Github Actions API. Please upload with the Codecov repository upload token to resolve issue.', code='not_found')}
```

**Review by CAGS295 — APPROVED** [review]

**Review by stjepangolemac — APPROVED** [review]

**Comment by stjepangolemac** [comment]
I'd just ignore the codecov failure. It seems to be a problem on their end.

**Review by MarvinJanssen — APPROVED** [review]

**Pull request merged**